modules/kubernetes-addons/main.tf

```diff
@@ -329,44 +329,92 @@ module "velero" {
 }
 
 module "opentelemetry_operator" {
-  count         = var.enable_opentelemetry_operator ? 1 : 0
+  count         = var.enable_amazon_eks_adot || var.enable_opentelemetry_operator ? 1 : 0
   source        = "./opentelemetry-operator"
-  helm_config   = var.opentelemetry_operator_helm_config
   addon_context = local.addon_context
+
+  # Amazon EKS ADOT add
+  enable_amazon_eks_adot = var.enable_amazon_eks_adot
+  addon_config = merge(
+    {
+      kubernetes_version = var.eks_cluster_version
+    },
+    var.amazon_eks_adot_config,
+  )
+
+  # Self-managed OpenTelemetry Operator via Helm chart
+  enable_opentelemetry_operator = var.enable_opentelemetry_operator
+  helm_config                   = var.opentelemetry_operator_helm_config
 }
 
 module "adot_collector_java" {
-  count                                = var.enable_adot_collector_java ? 1 : 0
-  source                               = "./adot-collector-java"
-  helm_config                          = var.adot_collector_java_helm_config
+  count  = var.enable_adot_collector_java ? 1 : 0
+  source = "./adot-collector-java"
+
+  helm_config = merge(
+    var.adot_collector_java_helm_config,
+    {
+      # Using the output from the operator addon, this will ensure correct deployment order
+      operator_namespace = module.opentelemetry_operator[0].namespace
```

Comment:
@bryantbiggs this doesn't produce a consistent behaviour, as it tries sometimes to install the app before the collector

Reply:
oy, ok - I guess we can abandon this and use the hard dependency with `depends_on`